lab2/README.md

```diff
@@ -0,0 +1,125 @@
+
+### Рабочий инстанс Nova
+
+Для создания рабочего инстанса необходимо перейти в `Вычислительные ресурсы->Инстансы`. Далее нажать на кнопку запустить инстанс, после чего появится диалоговое окно для настройки конфигурации. После того, как будут выбраны все параметры, запрос на создание виртуальной машины будет отправлен в Nova.
+![instance.png](./img/created_instance.png)
+
+Проверка IP-адреса виртуальной машины:
+![result-ip.png](./img/result_of_ip.png)
+
+## Выполнение задания
+
+### Задание
+
+1. Через Horizon создать еще одну приватную сеть+подсеть (адресация любая), подключить ее к уже существующему роутеру
+2. Через Openstack CLI сделать копию ранее созданного блочного устройства
+3. Через Openstack CLI создать еще одну ВМ (`openstack server create...`). Параметры машины:
+    - имя произвольное
+    - источник: созданная копия диска
+    - флейвор: созданный ранее (можно другой, если позволяют ресурсы ВМ)
+    - сеть: новосозданная приватная
+4. Через Horizon проверить, что все завелось
+
+### 1. Создание приватной сети через Horizon
+
+Для создания новой сети необходимо перейти в `Сеть->Сети` и нажать на кнопку создать сеть. После этого появится диалоговое окно, в котором можно настроить создание приватной сети, подсети, а также добавить детали для создания.
+
+![ui-network-creation.png](./img/ui_network_creation.png)
+
+В результате, в той же вкладке отобразится созданная сеть.
+
+![ui-networks](./img/ui_networks.png)
+
+### 2. Создание копии блочного устройства
+
+Команда `openstack create volume` поддерживает несколько источников создания дисков: image(--image), snapshot(--snapshot), volume(--source).
+
+Для создании из копии ранее созданного блочного устройства было использована два способа: snapshot и volume.
+
+Создание из snapshot: \
+![from-snapshot.png](./img/creation_disk_from_snapshot.png)
+
+Создание из volume: \
+![from-volume.png](./img/creation_block_from_volume.png)
+
+### 3. Создание виртуальной машины
+
+Для создания виртуальной машины использовались следующие аргументы: `--flavor`, `--volume`, `--network`, `--key-name`. \
+В итоге, создание полная команда выглядела следующим образом:
+![vm.png](./img/create_instance.png)
+
+Проверка созданного инстанса:
+![check-instance.png](./img/ui_check_created_instance.png)
+
+Проверка установленного IP-адреса:
+![ip-check.png](./img/result_of_ip_second.png)
+
+## Ответы на вопросы
+
+> Что именно сервис с помощью Keystone проверяет в токене пользователя, когда тот пытается осуществить операцию по отношению к этому сервису?
+
+Сервисы проверяют в Keystone две информации: корректный ли токен и разрешено ли использование сервиса.
+
+> При создании ВМ, Nova первым делом идет в Keystone, проверяет токен и т.д. Как думаете, к эндпоинту какого сервиса Nova идет следом?
+
+После того, как Nova проверит токен, следующим запросом является проверка наличии конфликтов с `nova-database` и при их отсутсвии создает начальную запись в базе данных для нового инстанса.
```

Comment:
```suggestion
После того, как Nova проверит токен, следующим запросом является проверка наличии конфликтов с `nova-database` и при их отсутсвии создает начальную запись в базе данных для нового инстанса. Однако следующим запросом в другой сервис, является запрос в Glance для получения образа вм.    
![Virtual-machine-spawning-sequence-in-OpenStack](https://github.com/ch4t5ky/2023_2024-openstack-teamwork/assets/141184937/337477a7-bf15-4932-a0a1-124443289ee7)
```

https://www.researchgate.net/publication/346347453_On_Construction_and_Performance_Evaluation_of_a_Virtual_Desktop_Infrastructure_With_GPU_Accelerated (https://www.researchgate.net/figure/Virtual-machine-spawning-sequence-in-OpenStack_fig3_346347453)